### PR TITLE
Share extension

### PR DESCRIPTION
**Pull request by Einh06**
**What's in this PR**
This PR allows the usage of the share extension with the app.
There are a lot of file moved, hence lots of modification, but the overall amount of change is small. Look at the ShareViewController in the extension folder to have an entry point to the share extension.

- [x] Needs release of `wire-ios-share-engine`
- [x] Needs release of `wire-ios-message-strategy`

### CODE REVIEW

File: ConfigurationSets/Wire-Share-Extension.base.xcconfig [hunk 1]
```diff
@@ -19,8 +19,8 @@
 
 WIRE_BUNDLE_DISPLAY_NAME = Wire
 SHARE_EXTENSION_BUNDLE_SUFFIX = extension-share
-SWIFT_OBJC_BRIDGING_HEADER = $(TARGET_NAME)/WireShareExtensionBridgingHeader.h
+//SWIFT_OBJC_BRIDGING_HEADER = $(TARGET_NAME)/WireShareExtensionBridgingHeader.h
```
Comment (mikeger) [inline ConfigurationSets/Wire-Share-Extension.base.xcconfig]: I guess it can be removed.

File: Wire-iOS Share Extension/Info.plist [hunk 2]
```diff
@@ -4,10 +4,12 @@
 <dict>
 	<key>AppIdentifierPrefix</key>
 	<string>${AppIdentifierPrefix}</string>
+	<key>ApplicationGroupIdentifier</key>
+	<string>group.${WIRE_BUNDLE_ID}</string>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
 	<key>CFBundleDisplayName</key>
-	<string>Wire</string>
+	<string>Wire-iOS Share Extension</string>
```
Comment (mikeger) [inline Wire-iOS Share Extension/Info.plist]: I would rename to "Wire Share Extension"

**Comment by daehn** [comment]
🔥 Conflicts

**Comment by mikeger** [comment]
The extension show empty conversation names for the conversations that don't have the name defined.

**Comment by mikeger** [comment]
It is possible to press "Post" several times, as after first press the UI gets "frozen" for some activity.

**Comment by daehn** [comment]
🔥 Conflicts

**Comment by daehn** [comment]
@Einh06 I merged and released `wire-ios-message-strategy` version 11.0.0

**Comment by daehn** [comment]
@Einh06 as the public API changed in the major release of `wire-ios-message-strategy` other frameworks which depend on it will also have to be updated.

**Comment by mikeger** [comment]
Also, sharing does not really work for me 😫

**Pull request merged**